src/PersistentOrderedMap.mo

```diff
@@ -227,6 +227,50 @@ module {
     public func contains<V>(m: Map<K, V>, key: K): Bool 
       = Internal.contains(m, compare, key);
 
+    /// Retrieves a key-value pair from the map `m` with a maximal key. If the map is empty returns `null`.
+    ///
+    /// Example:
+    /// ```motoko
+    /// import Map "mo:base/PersistentOrderedMap";
+    /// import Nat "mo:base/Nat";
+    /// import Debug "mo:base/Debug";
+    ///
+    /// let mapOps = Map.MapOps<Nat>(Nat.compare);
+    /// let map = mapOps.fromIter<Text>(Iter.fromArray([(0, "Zero"), (2, "Two"), (1, "One")]));
+    ///
+    /// Debug.print(debug_show(mapOps.maxEntry(map))); // => ?(2, "Two")
+    /// Debug.print(debug_show(mapOps.maxEntry(mapOps.empty()))); // => null
+    /// ```
+    ///
+    /// Runtime: `O(log(n))`
+    /// Space: `O(1)`.
+    /// where `n` denotes the number of key-value entries stored in the map and
+    /// assuming that the `compare` function implements an `O(1)` comparison.
```

Comment:
I think it doesn't require this assumption? We don't call `compare` here, just unconditionally take leftmost/rightmost element.

Reply:
fixed